core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -925,51 +933,195 @@ public boolean isHashValid(String hash) {
         }
     }
 
-    /* package */ static final JBCryptEncoder JBCRYPT_ENCODER = new JBCryptEncoder();
+    public static class PBKDF2PasswordEncoder implements PasswordHashEncoder {
+
+        public static final int HASH_LENGTH = 8;
+        public static final int ZERO = 0;
+        public static final int HEX_TWO = 2;
+
+        public static final int RADIX_LENGTH = 16;
+        public static final String STRING_SEPARATION = ":";
+
+        public static final int KEY_LENGTH = 512;
+        public static final String PBKDF2_ALGORITHM = "PBKDF2WithHmacSHA512";
+
+        @Override
+        public String encode(CharSequence rawPassword) {
+            return generatePasswordHashWithPBKDF2(rawPassword);
+        }
+
+        @Override
+        public boolean matches(CharSequence rawPassword, String encodedPassword) {
+                return validatePassword(rawPassword.toString(), encodedPassword);
+        }
+
+        @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "Accessible via System Groovy Scripts")
+        @Restricted(NoExternalUse.class)
+        private static int ITERATION_COUNT = SystemProperties.getInteger(HudsonPrivateSecurityRealm.class.getName() + ".ITERATION_COUNT", 10);
```

Comment:
Default should be 1000 if the format is changed to the latest. In the latest format we keep iteration_count explicite unlike  in BCrypt format: https://cloudbees.atlassian.net/wiki/spaces/CORE/pages/3796828182/PBKDF2+pattern+recognition+issue#Change-implementation